src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -305,17 +305,27 @@ private enum IsParamsValues : byte
                     }
                 }
 
-                var typeSymbol = DynamicTypeDecoder.TransformType(typeWithAnnotations.Type, countOfCustomModifiers, handle, moduleSymbol, refKind);
-                typeSymbol = NativeIntegerTypeDecoder.TransformType(typeSymbol, handle, moduleSymbol, containingSymbol.ContainingType);
-                typeWithAnnotations = typeWithAnnotations.WithTypeAndModifiers(typeSymbol, typeWithAnnotations.CustomModifiers);
-                // Decode nullable before tuple types to avoid converting between
-                // NamedTypeSymbol and TupleTypeSymbol unnecessarily.
-
-                // The containing type is passed to NullableTypeDecoder.TransformType to determine access
-                // for property parameters because the property does not have explicit accessibility in metadata.
-                var accessSymbol = containingSymbol.Kind == SymbolKind.Property ? containingSymbol.ContainingSymbol : containingSymbol;
-                typeWithAnnotations = NullableTypeDecoder.TransformType(typeWithAnnotations, handle, moduleSymbol, accessSymbol: accessSymbol, nullableContext: nullableContext);
-                typeWithAnnotations = TupleTypeDecoder.DecodeTupleTypesIfApplicable(typeWithAnnotations, handle, moduleSymbol);
+                if (moduleSymbol.TryDecodeExtensionErasureAttribute(handle, (PENamedTypeSymbol)containingSymbol.ContainingType, containingSymbol as PEMethodSymbol) is { } typeWithExtensions)
+                {
+                    typeWithAnnotations = TypeWithAnnotations.Create(typeWithExtensions);
```

Comment:
>typeWithAnnotations = TypeWithAnnotations.Create(typeWithExtensions); [](http://example.com/codeflow?start=20&length=69)

We might want to verify that our erasure process yields the type from the signature, otherwise IL that we emit at the call site won't match the method. Consider adding a prototype comment about that.